cmd/soroban-cli/src/commands/tx/update/sequence_number/next.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{
+    commands::{
+        global,
+        tx::xdr::{tx_envelope_from_input, Error as XdrParsingError},
+    },
+    config,
+    xdr::{self, SequenceNumber, TransactionEnvelope, WriteXdr},
+};
+
+#[derive(clap::Parser, Debug, Clone)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub config: config::Args,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    XdrStdin(#[from] XdrParsingError),
+    #[error(transparent)]
+    Xdr(#[from] xdr::Error),
+    #[error("only V1 transactions are supported")]
+    Unsupported,
+    #[error(transparent)]
+    RpcClient(#[from] crate::rpc::Error),
+    #[error(transparent)]
+    Config(#[from] config::Error),
+    #[error(transparent)]
+    Network(#[from] config::network::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let mut tx = tx_envelope_from_input(&None)?;
+        self.update_tx_env(&mut tx, global_args).await?;
+        println!("{}", tx.to_xdr_base64(xdr::Limits::none())?);
+        Ok(())
+    }
+
+    pub async fn update_tx_env(
+        &self,
+        tx_env: &mut TransactionEnvelope,
+        _global: &global::Args,
+    ) -> Result<(), Error> {
+        match tx_env {
+            TransactionEnvelope::Tx(transaction_v1_envelope) => {
+                let current_seq_num = self.current_seq_num().await?;
+                let next_seq_num = current_seq_num + 1;
+                transaction_v1_envelope.tx.seq_num = SequenceNumber(next_seq_num);
+            }
+            TransactionEnvelope::TxV0(_) | TransactionEnvelope::TxFeeBump(_) => {
+                return Err(Error::Unsupported);
+            }
+        };
+        Ok(())
+    }
+
+    async fn current_seq_num(&self) -> Result<i64, Error> {
+        let network = &self.config.get_network()?;
+        let client = network.rpc_client()?;
+        client
+            .verify_network_passphrase(Some(&network.network_passphrase))
+            .await?;
+
+        let muxed_account = self.config.source_account().await?;
```

Comment:
We should get source account from transaction, not the config.
I can create a transaction sourcing from alice, and may use any non-alice acount to bump the sequence.
Currently we will use wrong account to fetch the sequence (currently used account)

Reply:
good call! i made the update in [d3fdc92](https://github.com/stellar/stellar-cli/pull/1909/commits/d3fdc92f779c49b68c90598f9640488d1f379f68)